internal/plugin/files.go

```diff
@@ -14,6 +20,11 @@ func getConfigIfAny(pkg, projectDir string) (*config, error) {
 		return nil, errors.WithStack(err)
 	}
 
+	xdgRuntimePath, err := setupXdgRuntimePath(projectDir)
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Ack, on this being sub-optimal. I have a TODO below in line 65 to improve this.

`getConfigIfAny` is used in quite a few places:
```
❯ git grep -n getConfigIfAny
internal/plugin/files.go:17:func getConfigIfAny(pkg, projectDir string) (*config, error) {
internal/plugin/hooks.go:6:             c, err := getConfigIfAny(pkg, projectDir)
internal/plugin/info.go:12:     cfg, err := getConfigIfAny(pkg, projectDir)
internal/plugin/plugin.go:54:   cfg, err := getConfigIfAny(pkg, projectDir)
internal/plugin/plugin.go:136:          cfg, err := getConfigIfAny(pkg, projectDir)
internal/plugin/services.go:10:         conf, err := getConfigIfAny(pkg, projectDir)
```

So, not just `CreateFilesAndShowReadme`.

Reply:
moved TODO to this callsite.

Reply:
`getConfigIfAny` is a bit of a general load function for a single plugin, while the symlink is per-project. So IMO it makes sense to create in the same place we ensure virtenv directory is created (since that is what the symlink is supposed to point to)

Reply:
I can create it there, but I still need to pass it in. `GetConfigIfAny` is called from many different code paths. This is why I have added a TODO to improve the code more holistically in this respect.

Reply:
`CreateFilesAndShowReadme` gets called on every `ensurePackagesAreInstalled` so it's where you want this. 

It's not the best API, but I'm confident you will get the symlink you want.

Reply:
`CreateFilesAndShowReadme` creates the `virtenv` directory. It creates any support files needed for plugins (e.g configs, etc). Services cannot run without these files which is why I think it makes the most sense for us to create the symlink there.

The symlink path is computed deterministically using `projectDir` which means `GetConfigIfAny` can just get the `Symlink` path without having to try and re-create it.